docs/proposals/client-package-structure.md

```diff
@@ -0,0 +1,153 @@
+# Client: layering and package structure
+
+## Desired layers
+
+### Transport
+
+The transport layer is concerned with round-tripping requests to an apiserver
+somewhere. It consumes a Config object with options appropriate for this.
+(That's most of the current client.Config structure.)
+
+Third parties (e.g., open shift) may have their own implementation.
```

Comment:
I don't think we want Openshift or other clients to have their own implementation.  I believe transport is generic to "Kube-like" APIs - standard auth, security, behavior, and other patterns.  A kube like API should be able to use the Kube like transport